src/js/actions/transform.js

```diff
@@ -564,7 +564,10 @@ define(function (require, exports) {
                 return descriptor.endTransaction(transaction);
             })
             .then(function () {
-                return this.transfer(layerActions.resetIndex, document);
+                var boundsPromise = this.transfer(layerActions.resetBounds, document, document.layers.allSelected),
+                    indexPromise = this.transfer(layerActions.resetIndex, document);
```

Comment:
Just to be clear, do you not care which order these two actions are processed in? Although they might execute in an apparently deterministic order here, we shouldn't rely on that if there is some correctness constraint based on order. One reason why it might possibly matter is w.r.t. history.